### PR TITLE
Introduce sha256 security review fixes

### PR DESCRIPTION
**Pull request by AndrzejKurek**
Signed-off-by: Andrzej Kurek <andrzej.kurek@arm.com>

### CODE REVIEW

File: library/sha256.c [hunk 1]
```diff
@@ -317,47 +313,49 @@ int mbedtls_sha256_update_ret( mbedtls_sha256_context *ctx,
     SHA256_VALIDATE_RET( ctx != NULL );
     SHA256_VALIDATE_RET( ilen == 0 || input != NULL );
 
-    if( ilen == 0 )
+    if( ilen_dup == 0 )
```
Comment (shelib01) [inline library/sha256.c:316]: @AndrzejKurek , ilen_dup shouldn't be const ?

**Comment by AndrzejKurek** [comment]
This PR adds 36 bytes of codesize.

File: library/sha256.c (hunk 1 above)
Comment (piotr-now) [inline library/sha256.c:316]: I think it would be nice to add some comment about `ilen_dup` is used because of `volatile` and FI

Comment (AndrzejKurek) [reply library/sha256.c:317]: Added.

Comment (piotr-now) [reply library/sha256.c:317]: Thanks

**Pull request merged**